platform/server/elasticsearch/ElasticsearchService.ts

```diff
@@ -1,23 +1,27 @@
+import { Client } from 'elasticsearch';
 import {
   Observable,
   Subscription,
   k$,
   map,
   filter,
   switchMap,
-  shareLast
+  shareLast,
+  first,
+  toPromise
 } from '@elastic/kbn-observable';
 
-import { ElasticsearchClusterType } from './ElasticsearchConfig';
 import { ElasticsearchConfigs } from './ElasticsearchConfigs';
-import { Cluster } from './Cluster';
+import { AdminClient } from './AdminClient';
+import { ScopedDataClient } from './ScopedDataClient';
 import { LoggerFactory } from '../../logging';
 import { CoreService } from '../../types/CoreService';
+import { Headers } from '../http/Router/headers';
 
-type Clusters = { [type in ElasticsearchClusterType]: Cluster };
+type Clients = { data: Client, admin: Client };
```

Comment:
For now, let's make this:

```ts
type Clients = { [type in ElasticsearchClusterType]: Client };
```

(It will loop over every type in `ElasticsearchClusterType` and create an identical type to the one we already have for `Clients`. Then we can clean up or remove `ElasticsearchClusterType` later.)

(This is called a "Mapped type" in TypeScript, see https://www.typescriptlang.org/docs/handbook/advanced-types.html#mapped-types for more details about it)